src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/PreferStringContainsOverIndexOf.cs

```diff
@@ -0,0 +1,216 @@
+                    charAndComparisonTypeArgumentContainsMethod == null)
+                {
+                    return;
+                }
+
+                compilation.RegisterOperationBlockStartAction(context =>
+                {
+                    ISymbol symbol = context.OwningSymbol;
+                    if (!(context.OwningSymbol is IMethodSymbol method))
+                    {
+                        return;
+                    }
+                    context.RegisterOperationAction(context =>
+                    {
+                        ILocalReferenceOperation localReference = (ILocalReferenceOperation)context.Operation;
+                        ILocalSymbol symbol = localReference.Local;
+                        if (localSymbolsToNumberOfReferences.TryGetValue(symbol.Name, out int numberOfReferences))
+                        {
+                            localSymbolsToNumberOfReferences[symbol.Name] = ++numberOfReferences;
+                        }
+                        else
+                        {
+                            localSymbolsToNumberOfReferences.Add(symbol.Name, 1);
+                        }
+                    },
+                    OperationKind.LocalReference);
+
+                    context.RegisterOperationAction(context =>
+                    {
+                        IBinaryOperation blockOperation = (IBinaryOperation)context.Operation;
+                        if (blockOperation.OperatorKind != BinaryOperatorKind.Equals)
+                        {
+                            return;
+                        }
+                        // Check that the right hand side is a -1
+                        var rightOperand = blockOperation.RightOperand;
+                        if (rightOperand is IUnaryOperation unaryOperation && rightOperand.ConstantValue.HasValue && rightOperand.ConstantValue.Value is int intValue && intValue == -1)
+                        {
+                            var leftOperand = blockOperation.LeftOperand;
+                            var blockLocation = blockOperation.Syntax.GetLocation();
+                            if (leftOperand is ILocalReferenceOperation localReferenceOperation)
+                            {
+                                var variableName = localReferenceOperation.Local;
+                                SyntaxReference declaration = variableName.DeclaringSyntaxReferences.FirstOrDefault();
+                                if (declaration is null)
+                                {
+                                    return;
+                                }
+
+                                var semanticModel = context.Operation.SemanticModel;
+                                SyntaxNode declarationNode = declaration.GetSyntax(context.CancellationToken);
+                                var operation = semanticModel.GetOperation(declarationNode, context.CancellationToken);
```

Comment:
I might be wrong but this doesn't seem right to access syntax node then operation, I would expect you to be able to get this operation "directly".

Reply:
Any idea how? I'm not super familiar with the APIs here. I need to go from an `ILocalReferenceOperation` to it's `VariableDeclaratorOperation`?